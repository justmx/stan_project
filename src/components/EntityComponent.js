import React from 'react'
import PropTypes from 'prop-types'
import Grid from 'react-bootstrap/lib/Grid'
import Col from 'react-bootstrap/lib/Col'
import Row from 'react-bootstrap/lib/Row'
import '../styles/EntityComponent.scss'

const EntityComponent = ({ url, title }) => (
  <div id='title-component'>
    <img src={url} className='image'/>
    <div><span className='title'>{title}</span></div>
  </div>
)

EntityComponent.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string
}

export default EntityComponent
