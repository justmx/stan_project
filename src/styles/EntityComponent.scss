#title-component {
  margin-right: 20px;
  margin-top: 20px;
  margin-bottom: 20px;
  width: 200px;

  .image {
    width: 200px;
    height: 300px;
  }
  .title {
    font-size: 12px;
    color: black;
    font-weight: 600;
  }
}
