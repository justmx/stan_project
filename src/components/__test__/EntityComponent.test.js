import React from 'react'
import { shallow } from 'enzyme'
import EntityComponent from '../EntityComponent'

describe('Testing EntityComponent component', () => {
  it('renders correctly with url and title provided', () => {
    const wrapper = shallow(
      <EntityComponent
        title='abc'
        url='https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg'
      />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('#title-component').length).toBe(1)
  })

  it('renders correctly without url and title', () => {
    const wrapper = shallow(
      <EntityComponent />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
