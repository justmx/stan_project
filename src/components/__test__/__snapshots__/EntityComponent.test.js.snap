// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EntityComponent component renders correctly with url and title provided 1`] = `
<div
  id="title-component"
>
  <img
    className="image"
    src="https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg"
  />
  <div>
    <span
      className="title"
    >
      abc
    </span>
  </div>
</div>
`;

exports[`Testing EntityComponent component renders correctly without url and title 1`] = `
<div
  id="title-component"
>
  <img
    className="image"
    src="test-file-stub"
  />
  <div>
    <span
      className="title"
    />
  </div>
</div>
`;
